> > Add missing header and ifdefs to fix compilation of alarm counter for counter drivers.
> 
> Could you provide details of when it currently fails to compile? Ideally in a bug report, so we can backport the fix.

```
~/zephyrproject/zephyr/modules/openthread/platform/alarm_counter.c: In function 'otPlatTimeGetXtalAccuracy':
~/zephyrproject/zephyr/modules/openthread/platform/alarm_counter.c:177:16: error: implicit declaration of function 'otPlatRadioGetCslAccuracy'; did you mean 'otPlatTimeGetXtalAccuracy'? [-Wimplicit-function-declaration]
  177 |         return otPlatRadioGetCslAccuracy(NULL);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                otPlatTimeGetXtalAccuracy
~/zephyrproject/zephyr/modules/openthread/platform/alarm_counter.c: At top level:
~/zephyrproject/zephyr/modules/openthread/platform/alarm_counter.c:31:16: warning: 'time_offset_us' defined but not used [-Wunused-variable]
   31 | static int32_t time_offset_us;
      |                ^~~~~~~~~~~~~~
```
I am working on a 802.15.4 Radio driver to enable Thread operations on Silicon Labs EFR32 chipset. I encountered the above error when attempting to build an OpenThread shell application with OPENTHREAD_CSL_RECEIVER. Both the Radio Driver and the CSL support are coming in later PRs, but I figured I would fix this first.

--- a/modules/openthread/platform/alarm_counter.c
+++ b/modules/openthread/platform/alarm_counter.c
@@ -18,6 +18,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #include <openthread/platform/alarm-milli.h>
 #include <openthread/platform/alarm-micro.h>
+#include <openthread/platform/radio.h>
 #include <openthread-system.h>
 
 #include "platform-zephyr.h"
@@ -28,7 +29,9 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 void alarm_milli_set_time_offset_ms(int32_t offset_ms);
 
 static bool timer_us_fired;
+#if defined(CONFIG_OPENTHREAD_PLATFORM_PKT_TXTIME)
 static int32_t time_offset_us;
+#endif
 
 /* Wrap logic: split delays longer than one counter period into multiple alarms */
 typedef struct wrap_timer_data wrap_timer_data_t;

